Server/src/transport/plugin_hub.py

```
@@ -429,11 +454,77 @@ async def _handle_command_result(self, payload: CommandResultMessage) -> None:
+                    last_pong = cls._last_pong.get(session_id, 0)
+
+                # Check staleness: has it been too long since we got a pong?
+                elapsed = time.monotonic() - last_pong
+                if elapsed > cls.PING_TIMEOUT:
+                    logger.warning(
+                        f"[Ping] Session {session_id} stale: no pong for {elapsed:.1f}s "
+                        f"(timeout={cls.PING_TIMEOUT}s). Closing connection."
+                    )
+                    try:
+                        await websocket.close(code=1001)  # Going away
+                    except Exception as close_ex:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)